source/controlTypes.py

```diff
@@ -711,3 +714,18 @@ def processNegativeStates(role, states, reason, negativeStates):
 		# This is not a state change; only positive states were supplied.
 		# Return all negative states which should be spoken, excluding the positive states.
 		return speakNegatives - states
+
+def processAndLabelStates(role, states, reason, positiveStates, negativeStates, positiveStateLabelDict={}, negativeStateLabelDict={}):

```

Comment:
> why we both have states and positiveStates parameters for processPositiveStates

`states` contains the current states for the object. For REASON_CHANGE, `positiveStates` contains only those states that just got added. In other words, `positiveStates` contains only the states we want to possibly be reported at this time, but we might need the "unfiltered" states for the object, so we provide `states` as well.